### PR TITLE
Ft create post test

### PR DESCRIPTION
**Pull request by froggypirates22**
Added all the tests related to the us-create post.

-  DataMapper Test
    + Converting Model from Domain Model to Data Access Model 
    + Data Access Model to Domain Model
-  Controller Test -  PostAsync Method
-  Repository Test - AddAsync Method

### CODE REVIEW

File: Fakebook.Posts/Fakebook.Posts.DataAccess/PostsRepository.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+        public Task<Post> GetPostByEmail(string email)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public Task<User> GetUserByEmail(string email)
+        {
+            throw new System.NotImplementedException();
+        }
```
Comment (nodeJayS) [inline Fakebook.Posts/Fakebook.Posts.DataAccess/PostsRepository.cs:73]: Remove methods unrelated to the create-post user story

File: Fakebook.Posts/Fakebook.Posts.Domain/IPostsRepository.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Fakebook.Posts.Domain
+{
+    public interface IPostsRepository : ICollection<Post>, IAsyncEnumerable<Post>
+    {
+        ValueTask<Post> AddAsync(Post post);
+        Task<Post> GetPostByEmail(string email);
+        Task<User> GetUserByEmail(string email);
```
Comment (nodeJayS) [inline Fakebook.Posts/Fakebook.Posts.Domain/IPostsRepository.cs:11]: These should be in the see-posts user story, also remove the IAsyncEnumerable interface, we don't need it in this story

**Review by nodeJayS — COMMENTED** [review]
We will implement the other repo methods later, tests look ok otherwise

File: Fakebook.Posts/Fakebook.Posts.DataAccess/PostsRepository.cs (hunk 1 above)
Comment (froggypirates22) [reply Fakebook.Posts/Fakebook.Posts.DataAccess/PostsRepository.cs]: Removed methods not related to create-post.

File: Fakebook.Posts/Fakebook.Posts.Domain/IPostsRepository.cs (hunk 2 above)
Comment (froggypirates22) [reply Fakebook.Posts/Fakebook.Posts.Domain/IPostsRepository.cs]: Removed interface not related to create-post.

**Review by nodeJayS — APPROVED** [review]
LGTM

**Pull request merged**